go/vt/vtgate/gateway.go

```diff
@@ -35,8 +35,8 @@ import (
 // a query targeted to a keyspace/shard/tablet_type and send it off.
 
 var (
-	// GatewayImplementation allows you to choose which gateway to use for vtgate routing. Defaults to tabletgateway, other option is discoverygateway
-	GatewayImplementation = flag.String("gateway_implementation", "tabletgateway", "Allowed values: discoverygateway (deprecated), tabletgateway (default)")
+	_                     = flag.String("gateway_implementation", "", "Deprecated. Only tabletgateway is now supported, discoverygateway is no longer available")
+	GatewayImplementation = new(string)
```

Comment:
just make this `GatewayImplementation = &tabletGatewayImplementation` instead of assigning to it from an `init` function.

Reply:
That doesn't work because `tabletGatewayImplementation` is declared as a const.
```
go/vt/vtgate/gateway.go:39:26: cannot take the address of tabletGatewayImplementation
```

Reply:
I could change the type of `GatewayImplementation` from `string*` to `string` and change all references, but that will touch lots of files. I'm looking for the minimum necessary change.

Reply:
@deepthi I see, in that case the minimal change would be making a temp assignment right next to it.

```
var defaultGatewayImplementation = tabletGatewayImplementation
var GatewayImplementation = &defaultGatewayImplementation
```

That's the minimal amount of changes and doesn't allocate on heap or use `init`,